pkg/client/leaderelection/leaderelection.go

```diff
@@ -103,8 +103,8 @@ type LeaderElectionConfig struct {
 	// Identity is a unique identifier of the leader elector.
 	Identity string
 
-	EndpointsClient coreclientset.EndpointsGetter
-	EventRecorder   record.EventRecorder
+	Client        clientset.Interface
```

Comment:
> Also, since @timothysc is going to switch the lock to something different than endpoints (some third party object), we will need a wider interface anyway.

I prefer having it tighter, but I don't make an issue of it.

@timothysc thirdpartyresource for leader election?  Is there an issue for that?  I'm certainly for using something other than endpoints, but I think that third party resources may not be a great fit.